packages/flutter/lib/src/widgets/overscroll_indicator.dart

```diff
@@ -637,18 +644,282 @@ class _GlowingOverscrollIndicatorPainter extends CustomPainter {
-/// overscroll indication.
+/// A Material Design visual indication that a scroll view has overscrolled.
 ///
-/// To prevent the indicator from showing the indication, call [disallowGlow] on
-/// the notification.
+/// A [StretchingOverscrollIndicator] listens for [ScrollNotification]s in order
+/// to stretch the content of the [Scrollable]. These notifications are typically
+/// generated by a [ScrollView], such as a [ListView] or a [GridView].
+///
+/// When triggered, the [StretchingOverscrollIndicator] generates an
+/// [OverscrollIndicatorNotification] before showing an overscroll indication.
+/// To prevent the indicator from showing the indication, call
+/// [OverscrollIndicatorNotification.disallowIndicator] on the notification.
+///
+/// Created by [ScrollBehavior.buildOverscrollIndicator] on platforms
+/// (e.g., Android) that commonly use this type of overscroll indication when
+/// [ScrollBehavior.androidOverscrollIndicator] is true. Otherwise, the
+/// default [GlowingOverscrollIndicator] is applied.
+///
+/// See also:
+///
+///  * [OverscrollIndicatorNotification], which can be used to prevent the stretch
+///    effect from being applied at all.
+///  * [NotificationListener], to listen for the
+///    [OverscrollIndicatorNotification]
+class StretchingOverscrollIndicator extends StatefulWidget {
+  /// Creates a visual indication that a scroll view has overscrolled by
+  /// applying a stretch transformation to the content.
+  ///
+  /// In order for this widget to display an overscroll indication, the [child]
+  /// widget must contain a widget that generates a [ScrollNotification], such
+  /// as a [ListView] or a [GridView].
+  ///
+  /// The [axisDirection] and [notificationPredicate] arguments must not be null.
+  const StretchingOverscrollIndicator({
+    Key? key,
+    required this.axisDirection,
+    this.notificationPredicate = defaultScrollNotificationPredicate,
+    this.child,
+  }) : assert(axisDirection != null),
+        assert(notificationPredicate != null),
+        super(key: key);
+
+  /// {@macro flutter.overscroll.axisDirection}
+  final AxisDirection axisDirection;
+
+  /// {@macro flutter.overscroll.axis}
+  Axis get axis => axisDirectionToAxis(axisDirection);
+
+  /// {@macro flutter.overscroll.notificationPredicate}
+  final ScrollNotificationPredicate notificationPredicate;
+
+  /// The widget below this widget in the tree.
+  ///
+  /// The overscroll indicator will apply a stretch effect to this child. This
+  /// child (and its subtree) should include a source of [ScrollNotification]
+  /// notifications.
+  ///
+  /// Typically a [StretchingOverscrollIndicator] is created by a
+  /// [ScrollBehavior.buildOverscrollIndicator] method when opted-in using the
+  /// [ScrollBehavior.androidOverscrollIndicator] flag. In this case
+  /// the child is usually the one provided as an argument to that method.
+  final Widget? child;
```

Comment:
Curious, why this can be null?

Reply:
This is intentional, the pattern is the same for GlowingOverscrollIndicator.